simulator/modules/core/t/unit_test.cpp

```diff
@@ -92,14 +80,49 @@ TEST_CASE( "Perf_Sim: Register size")
 TEST_CASE( "Torture_Test: Perf_Sim")
```

Comment:
That one should be separated to different test cases for each architecture to avoid conflicts between different simulation instances.